geode-protobuf/src/main/java/org/apache/geode/protocol/protobuf/ProtobufSimpleAuthenticator.java

```diff
@@ -40,20 +42,28 @@ public void receiveMessage(InputStream inputStream, OutputStream outputStream,
     properties.setProperty("username", authenticationRequest.getUsername());
     properties.setProperty("password", authenticationRequest.getPassword());
 
+    authorizer = null; // authenticating a new user clears current authorizer
     try {
       Object principal = securityManager.authenticate(properties);
-      authenticated = principal != null;
+      if (principal != null) {
+        authorizer = new ProtobufSimpleAuthorizer(principal, securityManager);
```

Comment:
Why do we need a `ProtobufSimpleAuthorizer` per principal? Could we not have a single `Authorizer` that will authenticate all principals?